test/e2e_node/podresources_test.go

```diff
@@ -182,12 +183,17 @@ func matchPodDescWithResources(expected []podDesc, found podResMap) error {
 			return fmt.Errorf("no container resources for pod %q container %q", podReq.podName, podReq.cntName)
 		}
 
-		if podReq.cpuCount > 0 {
-			if len(cntInfo.CpuIds) != podReq.cpuCount {
-				return fmt.Errorf("pod %q container %q expected %d cpus got %v", podReq.podName, podReq.cntName, podReq.cpuCount, cntInfo.CpuIds)
+		cpuRequest := getCpuRequest(podReq.cpuRequest)
+		if cpuRequest > 0 && cpuRequest == float32(int(cpuRequest)) {
```

Comment:
for the sake of readability
// we have something to check
if cpuRequest > 0 {
   // 1. integral case of guranted pod
   your code
   // 2. non-integral case of all other QoS types
if you dont't like comment, just introduce a local helper if isIntegral(cpuRequest) ;)
}

Reply:
Went with the helper  ;)